### PR TITLE
Convert MesheryChart and MesheryChartDialog to functional component.

### LINKED ISSUE #8796: [UI] Convert `MesheryChart` and `MesheryChartDialog` from class based component to functional component
### Current Behavior
`MesheryChart` and `MesheryChartDialog` are class-based components

### Desired Behavior
`MesheryChart` and `MesheryChartDialog` should be functional components.

### Implementation
Convert all the Class-based components to new updated functional components. It is important here to convert a class-based component to a functional component you may need to convert some functions as well. Check if the function is used in other components or not.

Also, make sure to use the hooks while converting them. For example, you need to covert withNotify with useNotification, useRouter hook. Please check if a hook for functional components is available and use that.

---

### Contributor [Guides](https://docs.meshery.io/project/contributing) and [Handbook](https://layer5.io/community/handbook)

- 🎨 Wireframes and [designs for Meshery UI](https://www.figma.com/file/SMP3zxOjZztdOLtgN4dS2W/Meshery-UI) in Figma [(open invite)](https://www.figma.com/team_invite/redeem/qJy1c95qirjgWQODApilR9)
- 🖥 [Contributing to Meshery UI](https://docs.meshery.io/project/contributing/contributing-ui)
- 🙋🏾🙋🏼 Questions: [Discussion Forum](http://discuss.meshery.io) and [Community Slack](https://slack.meshery.io)

### PR DESCRIPTION
**Pull request by Aseem22**
This PR fixes #8796 

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

File: ui/components/MesheryChartDialog.js [hunk 1]
```diff
@@ -1,43 +1,43 @@
...
+import { makeStyles } from '@material-ui/core/styles';
 import Button from '@material-ui/core/Button';
 import Dialog from '@material-ui/core/Dialog';
 import DialogActions from '@material-ui/core/DialogActions';
 import DialogContent from '@material-ui/core/DialogContent';
 import DialogContentText from '@material-ui/core/DialogContentText';
 import DialogTitle from '@material-ui/core/DialogTitle';
 
-const styles = () => ({});
+const useStyles = makeStyles(() => ({
```
[the remark this answers is not in the document]
Comment (Aseem22) [reply ui/components/MesheryChartDialog.js:11]: I will check too

File: ui/components/MesheryChartDialog.js [hunk 2]
```diff
@@ -1,43 +1,43 @@
...
-              Close
-            </Button>
-          </DialogActions>
-        </Dialog>
-      </React.Fragment>
-    );
-  }
+function MesheryChartDialog(props) {
+  const { open, title, handleClose, content } = props;
+  const classes = useStyles();
```
[the remark this answers is not in the document]
Comment (leecalcote) [reply ui/components/MesheryChartDialog.js:17]: @Aseem22 are you still working incorporating this feedback?

[the remark this answers is not in the document]
Comment (Aseem22) [reply ui/components/MesheryChartDialog.js]: Yes @leecalcote, I implemented the feedbacks received from Antonette. Removing these empty styles doesn't seem to have any effect. I have pushed the changes too.

[the remark this answers is not in the document]
Comment (leecalcote) [reply ui/components/MesheryChartDialog.js]: Make sure that you respond to each and every piece of feedback, letting the maintainer know how you have addressed the feedback.

[the remark this answers is not in the document]
Comment (leecalcote) [reply ui/components/MesheryChartDialog.js]: Ask for the comment to be resolved or resolved the comment when the fix has been provided

[the remark this answers is not in the document]
Comment (Aseem22) [reply ui/components/MesheryChartDialog.js]: Sorry, I'll remember to do that from now on.

**Comment by Yashsharma1911** [comment]
Testing

**Comment by Aseem22** [comment]
> Testing

@Yashsharma1911 Any feedback from your test?

**Comment by Yashsharma1911** [comment]
> > Testing
> 
> @Yashsharma1911 Any feedback from your test?

I wasn't able to test at that time, testing now

File: ui/components/MesheryChartDialog.js [hunk 3]
```diff
@@ -1,50 +1,43 @@
 import React from 'react';
 import PropTypes from 'prop-types';
-import { withStyles } from '@material-ui/core/styles';
```
Comment (Yashsharma1911) [inline ui/components/MesheryChartDialog.js:3]: Thankyou for this

**Review by Yashsharma1911 — APPROVED** [review]
LGTM

**Comment by Yashsharma1911** [comment]
Thankyou @Aseem22

**Pull request merged**